gtk4/src/expression.rs

```diff
@@ -0,0 +1,362 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::{translate::*, ToValue};
+use glib::{IsA, Object, Type, Value};
+use std::boxed::Box as Box_;
+
+glib::wrapper! {
+    #[derive(Debug)]
+    pub struct Expression(Shared<ffi::GtkExpression>);
+
+    match fn {
+        ref => |ptr| ffi::gtk_expression_ref(ptr),
+        unref => |ptr| ffi::gtk_expression_unref(ptr),
+    }
+}
+
+impl glib::StaticType for Expression {
+    #[doc(alias = "gtk_expression_get_type")]
+    fn static_type() -> Type {
+        unsafe { from_glib(ffi::gtk_expression_get_type()) }
+    }
+}
+
+impl Expression {
```

Comment:
You probably want to add `downcast()` and `downcast_ref()` here too

Reply:
And a `get_type()`

Reply:
And `is()` too maybe

Reply:
> You probably want to add `downcast()` and `downcast_ref()` here too

The thing is I couldn't find a way to get the type or any information that would allow me to check whether the T::static_type() = self's type. That's the reason why I didn't add any of the those functions yet

Reply:
It's a `GTypeInstance` or not?

```cpp
struct _GTypeClass
{
  GType g_type;
};

struct _GTypeInstance
{
  GTypeClass *g_class;
};
```

Cast the pointer and get the type out of it :)

Reply:
Nice, thanks!